onnxruntime/test/testdata/transform/fusion/reshape_fusion_gen.py

```diff
@@ -245,13 +244,52 @@ def save_model(graph, file_name):
     ],
     [  # outputs
         helper.make_tensor_value_info('Result', TensorProto.FLOAT, [10, 20, 'unk']),
-        helper.make_tensor_value_info('div_out', TensorProto.INT64, []),
+        helper.make_tensor_value_info('slice_out', TensorProto.INT64, [1]),
     ],
     [  # initializers
         helper.make_tensor('indices0', TensorProto.INT64, [], [0]),
         helper.make_tensor('indices1', TensorProto.INT64, [], [1]),
-        helper.make_tensor('div_init', TensorProto.INT64, [], [1]),
+        helper.make_tensor('slice_starts', TensorProto.INT64, [1], [2]),
+        helper.make_tensor('slice_ends', TensorProto.INT64, [1], [3])        
     ]
 )
 
-save_model(graph, 'reshape_fusion_concat_subgraph_multiple_outputs.onnx')
\ No newline at end of file
+inferred_model = shape_inference.infer_shapes(helper.make_model(graph))
+onnx.save(inferred_model, 'reshape_fusion_concat_subgraph_multiple_outputs.onnx')
+
+graph = helper.make_graph(
+    [ # nodes
+        helper.make_node("Shape", ["SubgraphRoot"], ["shape0_out"], "shape0"),
+        helper.make_node("Shape", ["SubgraphRoot"], ["shape1_out"], "shape1"),
+        helper.make_node("Gather", ["shape0_out", "indices0"], ["gather0_out"], "gather0", axis=0),
+        helper.make_node("Gather", ["shape1_out", "indices1"], ["gather1_out"], "gather1", axis=0),
+        helper.make_node("Unsqueeze", ["gather0_out"], ["unsqueeze0_out"], "unsqueeze0", axes=[0]),
+        helper.make_node("Unsqueeze", ["gather1_out"], ["unsqueeze1_out"], "unsqueeze1", axes=[0]),
+        helper.make_node("Pad", ["unsqueeze0_out", "pads"], ["pad0_out"], "pad0", mode = "constant"),
+        helper.make_node("Pad", ["unsqueeze1_out", "pads"], ["pad1_out"], "pad1", mode = "constant"),
```

Comment:
This won't trigger the optimization as we are padding an extra value of "1" thus making tensor size > 1

Reply:
Do you need to pad all the inputs to the Concat node to not trigger the optimization or just one? 

---
In reply to: [439720318](https://github.com/microsoft/onnxruntime/pull/4185#discussion_r439720318) [](ancestors = 439720318)

Reply:
Now that I think about it, it is not needed. The mistake I did is to somehow think that Pad added a dimension which is why I padded the other inputs to keep the rank consistent feeding into Concat. Of course, this isn't correct. I ll remove the other pads.